src/phases.ts

```diff
@@ -1128,7 +1128,12 @@ export class EncounterPhase extends BattlePhase {
     case BattleSpec.FINAL_BOSS:
       const enemy = this.scene.getEnemyPokemon();
       this.scene.ui.showText(this.getEncounterMessage(), null, () => {
-        this.scene.ui.showDialogue(battleSpecDialogue[BattleSpec.FINAL_BOSS].encounter, enemy.species.name, null, () => {
+        const count = 5643853 + this.scene.gameData.gameStats.classicSessionsPlayed;
+        const ordinalUse = ["en", "es", "it"];
+        const ordinalIndex = (ordinalUse.includes(i18next.resolvedLanguage)) ? ["st","nd","rd"][((count+90)%100-10)%10-1] || "th" : "";
```

Comment:
```suggestion
        const ordinalIndex = (ordinalUse.includes(i18next.resolvedLanguage)) ? ["st", "nd", "rd"][((count + 90) % 100 - 10) % 10 - 1] || "th" : "";
```
Sorry, messed up the first comment/suggestion and some of the changes got left out.